src/main/java/de/hysky/skyblocker/skyblock/ChestValue.java

```diff
@@ -193,4 +232,58 @@ static Text getValueText(long value, boolean hasIncompleteData) {
 		UIAndVisualsConfig.ChestValue config = SkyblockerConfigManager.get().uiAndVisuals.chestValue;
 		return Text.literal(' ' + FORMATTER.format(value) + " Coins").formatted(hasIncompleteData ? config.incompleteColor : config.color);
 	}
+
+	private static void addValueToContainer(GenericContainerScreen genericContainerScreen, Text chestValue, Text title) {
+		Screens.getButtons(genericContainerScreen).removeIf(clickableWidget -> clickableWidget instanceof ChestValueTextWidget);
+		int backgroundWidth = ((HandledScreenAccessor) genericContainerScreen).getBackgroundWidth();
+		int y = ((HandledScreenAccessor) genericContainerScreen).getY();
+		int x = ((HandledScreenAccessor) genericContainerScreen).getX();
+		((ScreenAccessor) genericContainerScreen).setTitle(Text.empty());
+		TextRenderer textRenderer = MinecraftClient.getInstance().textRenderer;
+		int chestValueWidth = Math.min(textRenderer.getWidth(chestValue), Math.max((backgroundWidth - 8) / 2 - 2, backgroundWidth - 8 - textRenderer.getWidth(title)));
+
+		TextWidget chestValueWidget = new ChestValueTextWidget(chestValueWidth, textRenderer.fontHeight, chestValue, textRenderer);
+		chestValueWidget.setPosition(x + backgroundWidth - chestValueWidget.getWidth() - 4, y + 6);
+		Screens.getButtons(genericContainerScreen).add(chestValueWidget);
+
+		ChestValueTextWidget chestTitleWidget = new ChestValueTextWidget(backgroundWidth - 8 - chestValueWidth - 2, textRenderer.fontHeight, title.copy().fillStyle(Style.EMPTY.withColor(4210752)), textRenderer);
```

Comment:
Wait I’m confused. Why are you rendering the title? Vanilla does that already?

Reply:
to shrink the title if need be, cuz if the title already takes 95% of the title space then you won't be seeing the coins x)
to see what i mean test in an enchanted sack or check the video